src/librustc/middle/free_region.rs

```diff
@@ -10,24 +10,18 @@
 
 //! This file defines
 
+use middle::ty::{self, FreeRegion, Region};
 use middle::wf::ImpliedBound;
-use middle::ty::{self, FreeRegion};
-use util::common::can_reach;
-use util::nodemap::{FnvHashMap, FnvHashSet};
+use rustc_data_structures::transitive_relation::TransitiveRelation;
 
 #[derive(Clone)]
 pub struct FreeRegionMap {
-    /// `map` maps from a free region `a` to a list of
-    /// free regions `bs` such that `a <= b for all b in bs`
-    map: FnvHashMap<FreeRegion, Vec<FreeRegion>>,
-    /// regions that are required to outlive (and therefore be
-    /// equal to) 'static.
-    statics: FnvHashSet<FreeRegion>
+    relation: TransitiveRelation<Region>
```

Comment:
I know the code is practically self-documenting, but it might be nice to at least state here what relation is encoded in this field.  (Namely whether a relation `a -> b` is interpreted as `a: b` or as `b: a`.)